source/array_table_model.ts

```diff
@@ -1,33 +1,67 @@
 import * as Kola from 'kola-signals';
-import { Operation } from './operations';
-import { TableModel } from './table_model';
+import {AddRowOperation, MoveRowOperation, Operation, RemoveRowOperation,
+  Transaction, UpdateOperation} from './operations';
+import {TableModel} from './table_model';
 
 /** Implements a TableModel using an 2-dimensional array. */
 export class ArrayTableModel extends TableModel {
 
   /** Constructs an empty model. */
   constructor() {
     super();
-    this.dispatcher = new Kola.Dispatcher<Operation>();
+    this.dispatcher = new Kola.Dispatcher<Operation | Transaction>();
+    this.table = [];
+    this.transactionArray = [];
+    this.transactionDepth = -1;
   }
 
   /**
    * Marks the beginning of a transaction. In cases where a transaction is
    * already being processed, then the sub-transaction gets consolidated into
    * the parent transaction.
    */
-  public beginTransaction(): void {}
+  public beginTransaction(): void {
+    if(this.transactionDepth > -1) {
+      const lastNestedArray = this.getLastNestedArray(this.transactionArray,
+        this.transactionDepth);
+      lastNestedArray.push([]);
+    }
+    this.transactionDepth += 1;
+  }
 
   /** Ends a transaction. */
-  public endTransaction(): void {}
+  public endTransaction(): void {
+    if(this.transactionDepth > 0) {
+      const lastNestedArrayParent = this.getLastNestedArray(
+        this.transactionArray, this.transactionDepth - 1);
+      const lastNestedArray = lastNestedArrayParent.pop();
+      const transaction = new Transaction(lastNestedArray);
+      lastNestedArrayParent.splice(lastNestedArrayParent.length, 0,
+        ...lastNestedArray);
+      this.transactionDepth -= 1;
+      this.dispatcher.dispatch(transaction);      
+    } else if(this.transactionDepth === 0){
+      const transaction = new Transaction(this.transactionArray);
+      this.transactionArray = [];
+      this.transactionDepth -= 1;
+      this.dispatcher.dispatch(transaction);
+    }
+  }
 
   /**
    * Appends a row to the table.
    * @param row - The row to append.
    * @throws RangeError - The length of the row being added is not exactly equal
    *                      to this table's columnCount.
    */
-  public push(row: any[]): void {}
+  public push(row: any[]): void {
+    if(row.length > this.columnCount && this.columnCount !== 0) {
+      throw new RangeError('The length of the row being added is not ' +
+        'exactly equal to this table\'s columnCount.');
+    }
+    this.table.push(row.slice());
+    this.processOperation(new AddRowOperation(this.table.length - 1));
```

Comment:
It would be even better to reuse ```insert``` to implement this to avoid duplicating all the issues.